File: backend/app/schemas/pedido.py
# backend/app/schemas/pedido.py
from pydantic import BaseModel, EmailStr
from typing import Optional, List
from uuid import UUID
from datetime import datetime, date
from decimal import Decimal
from enum import Enum

class PedidoStatus(str, Enum):
    aguardando_aprovacao = 'Aguardando Aprovação'
    aprovado = 'Aprovado'
    em_andamento = 'Em Andamento'
    aguardando_pagamento = 'Aguardando Pagamento'
    concluido = 'Concluido'
    cancelado = 'Cancelado'

class ItemPedidoBase(BaseModel):
    descricao: str
    quantidade: Decimal
    unidade: str
    preco_unitario: Decimal
    observacoes: Optional[str] = None

class ItemPedidoCreate(ItemPedidoBase):
    pass

class ItemPedidoUpdate(ItemPedidoBase):
    pass

class ItemPedidoRead(ItemPedidoBase):
    id: UUID
    pedido_id: UUID
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True

class PedidoBase(BaseModel):
    cliente_id: UUID
    numero: str
    status: Optional[PedidoStatus] = PedidoStatus.aguardando_aprovacao
    referencia: Optional[str] = None
    validade_orcamento_dias: Optional[int] = 30
    prazo_execucao_dias: Optional[int] = None
    data_inicio: Optional[date] = None
    data_fim_prevista: Optional[date] = None
    data_fim_real: Optional[date] = None
    valor_subtotal: Optional[Decimal] = 0
    valor_desconto: Optional[Decimal] = 0
    valor_total: Optional[Decimal] = 0

class PedidoCreate(PedidoBase):
    itens: List[ItemPedidoCreate] = []

class PedidoUpdate(PedidoBase):
    itens: Optional[List[ItemPedidoUpdate]] = None # Para permitir atualizar itens separadamente

class PedidoRead(PedidoBase):
    id: UUID
    tenant_id: UUID
    created_at: datetime
    updated_at: datetime
    itens: List[ItemPedidoRead] = []

    class Config:
        from_attributes = True

File: frontend/src/types/pedido.ts
// frontend/src/types/pedido.ts
//import { Decimal } from 'decimal.js'; // Não é necessário importar decimal.js para tipos simples
import { ClienteRead } from './cliente';

export enum PedidoStatus {
  AguardandoAprovacao = 'Aguardando Aprovação',
  Aprovado = 'Aprovado',
  EmAndamento = 'Em Andamento',
  AguardandoPagamento = 'Aguardando Pagamento',
  Concluido = 'Concluido',
  Cancelado = 'Cancelado',
}

export interface ItemPedidoBase {
  descricao: string;
  quantidade: number;
  unidade: string;
  preco_unitario: number;
  observacoes?: string;
}

export interface ItemPedidoCreate extends ItemPedidoBase {}

export interface ItemPedidoUpdate extends ItemPedidoBase {
  id?: string; // ID opcional para atualização de itens existentes
}

export interface ItemPedidoRead extends ItemPedidoBase {
  id: string;
  pedido_id: string;
  created_at: string;
  updated_at: string;
}

export interface PedidoBase {
  cliente_id: string;
  numero: string;
  status?: PedidoStatus;
  referencia?: string;
  validade_orcamento_dias?: number;
  prazo_execucao_dias?: number;
  data_inicio?: string; // Date em backend, string no frontend (ISO 8601)
  data_fim_prevista?: string;
  data_fim_real?: string;
  valor_subtotal?: number;
  valor_desconto?: number;
  valor_total?: number;
}

export interface PedidoCreate extends PedidoBase {
  itens?: ItemPedidoCreate[];
}

export interface PedidoUpdate extends PedidoBase {
  itens?: ItemPedidoUpdate[];
}

export interface PedidoRead extends PedidoBase {
  id: string;
  tenant_id: string;
  created_at: string;
  updated_at: string;
  itens: ItemPedidoRead[];
  cliente?: ClienteRead; // Incluído para completar a estrutura, como comentado
}
